### PR TITLE
Amls 4885 - Page split for previous name question

### PR DESCRIPTION
**Pull request by slavvco**
As above and in ticket. New views, controllers, updated model, specs.

## Related / Dependant PRs?

- none

## How Has This Been Tested?

- manually

## Checklist

- [x] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [ ] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [x] Requires acceptance test run.
- [ ] Appropriate labels added.
- [x] RELEASE NOTES ADDED.

### CODE REVIEW

File: app/controllers/responsiblepeople/LegalNameInputController.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  def post(index: Int, edit: Boolean = false, flow: Option[String] = None) = authAction.async {
+      implicit request => {
+        Form2[PreviousName](request.body) match {
+          case f: InvalidForm =>
+            getData[ResponsiblePerson](request.credId, index) map { rp =>
+              BadRequest(views.html.responsiblepeople.legal_name_input(f, edit, index, flow, ControllerHelper.rpTitleName(rp)))
+            }
+          case ValidForm(_, data) => {
+            for {
+              _ <- {
```
Comment (ajchilts) [inline app/controllers/responsiblepeople/LegalNameInputController.scala]: Additional braces here? Do we need em?

File: app/models/responsiblepeople/PreviousName.scala [hunk 2]
```diff
@@ -16,53 +16,47 @@
...
-        (
-          (__ \ "firstName").read(optionR(genericNameRule("error.required.rp.first_name"))) ~
-            (__ \ "middleName").read(optionR(genericNameRule())) ~
-            (__ \ "lastName").read(optionR(genericNameRule("error.required.rp.last_name")))
-          ).tupled andThen iR).map(t => PreviousName(Some(true), t._1, t._2, t._3))
-      case false => Rule.fromMapping { _ => Valid(PreviousName(Some(false), None, None, None)) }
+        (__ \ "firstName").read(firstNameRule) ~
+          (__ \ "middleName").read(optionR(middleNameRule)) ~
+          (__ \ "lastName").read(lastNameRule)
+        ).tupled.map(t => PreviousName(Some(true), Some(t._1), t._2, Some(t._3)))
```
Comment (ajchilts) [inline app/models/responsiblepeople/PreviousName.scala]: Can we have a more descriptive variable name than 't' here?

File: app/views/responsiblepeople/detailed_answers.scala.html [hunk 3]
```diff
@@ -115,30 +115,45 @@
                 @checkYourAnswersRow(
                     question = Messages("responsiblepeople.legalName.heading", personName),
                     editUrl = controllers.responsiblepeople.routes.LegalNameController.get(idx, true, flow).toString(),
-                    editLinkTag = "rp-legalname-edit"
+                    editLinkTag = "rp-legalname-yes-no-edit"
+                ) {
+                    @ln.hasPreviousName match {
+                        case Some(false) => {
+                            <p>@Messages("lbl.no")</p>
```
Comment (ajchilts) [inline app/views/responsiblepeople/detailed_answers.scala.html]: We can remove the <p> tags as only needed where two answers are output in the same panel.

File: app/views/responsiblepeople/detailed_answers.scala.html [hunk 4]
```diff
@@ -405,10 +420,10 @@
                             </ul>
                         }
                     }
-                    <p>@ex match {
-                    case ExperienceTrainingYes(info) => {<p>@Messages("lbl.yes")</p> @info}
+                    @ex match {
+                        case ExperienceTrainingYes(info) => {<p>@Messages("lbl.yes")</p> <p>@info</p>}
```
Comment (ajchilts) [inline app/views/responsiblepeople/detailed_answers.scala.html]: Too many <p> elements? (Or is this the one we resolved earlier?)

Comment (matthewharris1) [reply app/views/responsiblepeople/detailed_answers.scala.html]: This is the one we resolved earlier

File: app/controllers/responsiblepeople/LegalNameInputController.scala (hunk 1 above)
Comment (matthewharris1) [reply app/controllers/responsiblepeople/LegalNameInputController.scala]: Removed

File: app/models/responsiblepeople/PreviousName.scala (hunk 2 above)
Comment (matthewharris1) [reply app/models/responsiblepeople/PreviousName.scala]: Done

File: app/views/responsiblepeople/detailed_answers.scala.html (hunk 3 above)
Comment (matthewharris1) [reply app/views/responsiblepeople/detailed_answers.scala.html]: Removed

**Pull request merged**